fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/NetworkLogger.java

```diff
@@ -62,52 +67,19 @@ public static void loadFromPersistPreference(Application application) {
             .getString(context.getString(R.string.settings_server_url_key), ""));
 
     // Register the callback for a graceful shutdown
-    application.registerActivityLifecycleCallbacks(
-        new Application.ActivityLifecycleCallbacks() {
-
-          private static final boolean SAVE_ENERGY = false;
-
-          @Override
-          public void onActivityCreated(
-              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
-            // Nothing
-          }
-
-          @Override
-          public void onActivityStarted(@NonNull Activity activity) {
-            // Nothing
-          }
-
-          @Override
-          public void onActivityResumed(@NonNull Activity activity) {
-            // Nothing
-          }
-
-          @Override
-          public void onActivityPaused(@NonNull Activity activity) {
-            if (SAVE_ENERGY) {
-              closeWebSocket();
-            }
-          }
-
-          @Override
-          public void onActivityStopped(@NonNull Activity activity) {
-            if (SAVE_ENERGY) {
-              closeWebSocket();
-            }
-          }
-
-          @Override
-          public void onActivitySaveInstanceState(
-              @NonNull Activity activity, @NonNull Bundle outState) {
-            // Nothing
-          }
-
-          @Override
-          public void onActivityDestroyed(@NonNull Activity activity) {
+    Map<Lifecycle.Event, Consumer<Activity>> consumerMap = new EnumMap<>(Lifecycle.Event.class);
+    consumerMap.put(Lifecycle.Event.ON_DESTROY, activity -> closeWebSocket());
+    Consumer<Activity> saverConsumer =
+        activity -> {
+          if (SAVE_ENERGY) {
+            closeWebSocket();
             closeWebSocket();
           }
-        });
```

Comment:
I'd add a comment explaining there's nothing to be done if not in Energy Saving mode